docs/docs/concepts/structure/users.md

```diff
@@ -15,39 +16,65 @@ Human users typically logon with an interactive login.
 This means that an application redirects a user to a website ("login page") where the user can provide the credentials.
 ZITADEL handles the authentication and provides the application with a token that verifies the authentication process.
 
+Read more on how to [login users with ZITADEL](/docs/guides/integrate/login/login-users).
+
 ### Service users
 
 Service users are for machine-to-machine communication and you would use those typically to access secure backend services.
 For example in ZITADEL you would require an authenticated Service User to access the Management API.
 The main difference between human and machine users is the type of credentials that can be used for authentication: Human users typically logon via an login prompt, but Machine users require a non-interactive logon process.
 
+Learn how to [use service users](/docs/guides/integrate/service-users/authenticate-service-users) with ZITADEL.
+
 ### Managers
 
 Any user, human or service user, can be given a [Manager](/concepts/structure/managers) role.
 Given a manager role, a user is not only an end-user of ZITADEL but can also manage certain aspects of ZITADEL itself.
 
-## Constraints
+### Federated users
+
+Federated users are identities that are managed by a third-party identity provider.
+Users can login via an external identity provider, using [identity brokering](../features/identity-brokering) ("Single-sign-on").
+Federated user [accounts are linked](../features/account-linking) to internal users to be able to assign roles and keep an audit trail.
+
+### External users
+
+In a multi-tenancy architecture, you might use [organizations](organizations) to separate user groups.
+By using [external user grants](../features/external-user-grant) an organization is able to invite users from another organization.
+These invited users are called external users.
+
+## Considerations
+
+### Uniqueness of users
 
 Users can only exist within one [organization](/concepts/structure/organizations).
 It is currently not possible to move users between organizations.
 
 User accounts are uniquely identified by their `id` or `loginname` in combination of the `organization domain` (eg, `road.runner@acme.zitadel.local`).
 You can use the same email address for different user accounts.
 
-## Where to store users
+### How to structure user pools
+
+Consider this general recommendation as a starting point:
+
+- Create one organization ("default organization") for your own team
+- Configure projects and applications in the default organization
+- Structure users in the smallest unit of self-managed groups ("teams", "user pools")
```

Comment:
Do we really recommend this generally? Problem I see when you add teams from a company, you can only have the domain once for an organization.
e.g if you have company-a and the team marketing and engineering, you can put them all in the same or or in different. but only in the same org you will be able to add theverified domain. 
I would not generally recommend adding the teams. but there might be use cases where it makes sense.

Reply:
@hifabienne sorry with team I meant for the own company. I will re-write that. We call an organization in the cloud also Team... so I wanted to get closer to that. But I see that this is confusing. We should also keep that in mind for the cloud.

Reply:
I've also added a consideration below. For me this is still the basic recommendation and then from there on the users need to figure out why this would not work in their case (like you just did). We need to provide a better series here anyways. 

FYI - @dakshitha please see comments from Fabi here on the architecture.